src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/UIPropertyDataProducer.cs

```diff
@@ -106,8 +106,15 @@ public static IEntityValue CreateUIPropertyValue(IQueryExecutionContext queryExe
 
             if (requestedProperties.VisibilityCondition)
             {
-                string? visibilityCondition = property.GetMetadataValueOrNull("VisibilityCondition");
-                newUIProperty.VisibilityCondition = visibilityCondition ?? string.Empty;
+                if (!property.Visible)
```

Comment:
Could you please clarify why the Visible property remains? What is it visible to? The front end?

Reply:
The data model that the back end exposes to the client doesn't have `Visible`, only `VisibilityCondition`. The `Visible` property is part of the general-purpose properties system implemented by CPS, and the back-end sits on top of that.

The end result of all this is that the property is available to the client code, but it won't be shown to the user.

Reply:
I see, so we want the property to always be available to the client but with the correct VisibilityCondition. Thank you for the clarification